pkgs/development/python-modules/pyscf/default.nix

```
@@ -0,0 +1,164 @@
+{ lib
+, buildPythonPackage
+, isPy27
+, fetchFromGitHub
+# C/build dependencies
+, cmake
+, openblas
+, libcint
+, libxc
+, xcfun
+# Python dependencies
+, h5py
+, numpy
+, scipy
+  # Check Inputs
+, nose
+, nose-exclude
+}:
+
+buildPythonPackage rec {
+  pname = "pyscf";
+  version = "1.7.1";
+
+  # must download from GitHub to get the Cmake & C source files
+  src = fetchFromGitHub {
+    owner = "pyscf";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0fciw9id8fr9396sz52ap5gys0i9hmrmrwm0i1k1k7aciqab3kls";
+  };
+
+  disabled = isPy27;
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [
+    libcint
+    libxc
+    openblas
+    xcfun
+  ];
+
+  postPatch = ''
+    mkdir -p ./pyscf/lib/deps/include ./pyscf/lib/deps/lib
+    ln -s ${lib.getDev xcfun}/include/xc.h ./pyscf/lib/deps/include/xcfun.h
+    ln -s ${lib.getLib xcfun}/lib/libxc.so ./pyscf/lib/deps/lib/libxcfun.so
+    substituteInPlace pyscf/rt/__init__.py --replace "from tdscf import *" "from pyscf.tdscf import *"
+  '';
+
+
+  cmakeFlags = [
+    # disable rebuilding/downloading the required libraries
+    "-DBUILD_LIBCINT=0"
+    "-DBUILD_LIBXC=0"
+    "-DBUILD_XCFUN=0"
+    "-DENABLE_XCFUN=1"
+  ];
+  # Configure CMake to build C files in pyscf/lib. Python build expects files in ./pyscf/lib/build
+  preConfigure = ''
+    pushd pyscf/lib
+  '';
+  postConfigure = ''
+    popd
+  '';
+
+  # Build C dependencies, then build python package.
+  preBuild = ''
+    pushd pyscf/lib/build
+    make
+    popd
+  '';
```

this should work as well
```suggestion
  makeFlags = [ "-C pyscf/lib/build" ];
```